runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -1368,8 +1368,12 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(
 					target = (jlong)offset;
 				}
 			}
-
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			/* In project Valhalla fields may start at offset 0. */
```

Comment:
This should be restructured so there's only one `if` statement, not an extra `{` that would be unmatched, and eliminate duplication of the common subexpression:
```c
			if (
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
				/* In project Valhalla fields may start at offset 0. */
				(NULL != new_clazz)
#else /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
				(0 != target)
#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
				&& ((0 != vmindex) || J9_ARE_ANY_BITS_SET(flags, MN_IS_METHOD | MN_IS_CONSTRUCTOR))
			) {
```

Reply:
I updated the change to match this.